data_catalog_backend/routes/admin/category_routes.py

```diff
@@ -34,3 +35,26 @@ async def add_category(
     except Exception as e:
         logger.error(e)
         raise HTTPException(status_code=500, detail=str(e))
+
+
+@router.delete(
+    "/{id}",
+    description="Delete a category",
+    response_model=CategoryResponse,
+    response_model_exclude_none=True,
+    tags=["admin"],
+)
+async def delete_category(
+    category_id: uuid.UUID,
+    current_user: Annotated[User, Depends(authenticate_user)],
+    service: CategoryService = Depends(get_category_service),
+) -> CategoryResponse:
+    deleted_category = service.delete_category(category_id, current_user)
+    if not deleted_category:
+        raise HTTPException(status_code=404, detail="Category not found")
+    if deleted_category.resources and len(deleted_category.resources) > 0:
+        raise HTTPException(
+            status_code=400, detail="Cannot delete category with existing resources"
+        )
+    converted = CategoryResponse.model_validate(deleted_category)
+    return converted
```

Comment:
In a delete endpoint, the http status for a successful deletion, should be 204 No Content.
Now, it returns 200 with the content that no longer exists.